docs/mgmt/readme.md

```diff
@@ -0,0 +1,658 @@
+
+### `[Resource]Data` class
+
+This class represents the "model" of the resource, and it corresponds to a model definition in the swagger, usually with the same name by trimming the word `Data` off.
+
+### `[Resource]Resource` class
+
+This class represents the a full resource client object which contains a `Data` property exposing the details as a `[Resource]Data` type.
+
+### `[Resource]Collection` class
+
+This class represents the operations you can perform on a collection of resources belonging to a specific parent resource. This object provides most of the logical collection operations.
+
+| Collection Behavior | Collection Method |
+| :--- | :--- |
+| Iterate/List | `GetAll()` |
+| Index | `Get(string name)` |
+| Add | `CreateOrUpdate(string name, [Resource]Data data)` |
+| Contains | `Exists(string name)` |
+| TryGet | `GetIfExists(string name)` |
+
+The `[Resource]Collection` class is designed to implement `IEnumerable<[Resource]Resource>` and `IAsyncEnumerable<[Resource]Resource>` interfaces, therefore the `GetAll()` method with no required parameters are required on this class for the generator. In rare cases, this rule is violated and a `list-exception` configuration will be required to temporarily solve this. See see [List exception](#list-exception) section for details.
+
+The parent resource of this resource will carry all the methods to get its child resources by returning its corresponding collection. For instance, on the resource of `VirtualNetworkResource`, you will find a method `GetSubnets` which returns a `SubnetCollection` to represent a collection of `SubnetResource`:
+```csharp
+public partial class VirtualNetworkResource : ArmResource
+{
+    public virtual SubnetCollection GetSubnets()
+    {
+        /* ... */
+    }
+}
+```
+
+For the cases that the parent resource is not in the same RP, the "get child resource method" will be put in an extension class. For instance, the parent resource of `VirtualNetworkResource` is the resource group, and you will find an extension method of `ResourceGroupResource` in the `Azure.ResourceManager.Network` package returning `VirtualNetworkCollection` like this:
+```csharp
+public static partial class NetworkExtensions
+{
+    public static VirtualNetworkCollection GetVirtualNetworks(this ResourceGroupResource resourceGroup)
+    {
+        /* ... */
+    }
+}
+```
+
+## How does the generator identify a resource
+
+The management .Net SDK generates the resources with hierarchy, therefore the code generator makes quite a few modification on the code model input from the `modelerfour`. To better recognize the hierarchical structure of resources, the the code generator will generate everything from the point of view of request paths, instead of operation groups. Therefore quite a few new configurations are introduced to tweak the behavior how the generator identifies the resource classes as well as the hierarchy.
```

Comment:
```suggestion
The management .Net SDK generates the resources with hierarchy, therefore the code generator makes quite a few modification on the code model input from the `modelerfour`. To better recognize the hierarchical structure of resources, the code generator will generate everything from the point of view of request paths, instead of operation groups. Therefore quite a few new configurations are introduced to tweak the behavior how the generator identifies the resource classes as well as the hierarchy.
```